packages/aws-client/src/s3-jitter.ts

```diff
@@ -0,0 +1,31 @@
+import Logger from '@cumulus/logger';
+
+const log = new Logger({ sender: 's3-jitter' });
+
+/**
+ * Introduces random jitter delay to stagger concurrent S3 operations.
+ * This helps prevent AWS S3 SlowDown errors when many operations occur simultaneously.
+ *
+ * @param maxJitterMs - Maximum jitter time in milliseconds (0-59000).
+ *   If 0, no delay is applied.
+ * @param operation - Optional operation name for logging context
+ * @returns A Promise that resolves after the random delay
+ */
+export const applyS3Jitter = async (
+  maxJitterMs: number,
+  operation?: string
+): Promise<void> => {
+  if (maxJitterMs <= 0) {
+    //log.debug('S3 jitter disabled (maxJitterMs = 0)');
+    return;
+  }
+
+  const jitterMs = Math.floor(Math.random() * maxJitterMs);
+
+  const logContext = operation ? ` for ${operation}` : '';
+  log.info(`Applying S3 jitter: ${jitterMs}ms${logContext} (max: ${maxJitterMs}ms)`);
+
+  await new Promise((resolve) => setTimeout(resolve, jitterMs));
```

Comment:
just because we have this shorthand already written
```suggestion
import { sleep } from '@cumulus/common';
  await sleep(jitterMs);
```

Reply:
nice, I replaced the setTimeout with our sleep() function from common.